canal/canal_test.go

```diff
@@ -136,98 +136,138 @@ func (s *canalTestSuite) TestCanalFilter(c *C) {
 
 func TestCreateTableExp(t *testing.T) {
 	cases := []string{
-		"CREATE TABLE `mydb.mytable` (`id` int(10)) ENGINE=InnoDB",
-		"CREATE TABLE `mytable` (`id` int(10)) ENGINE=InnoDB",
-		"CREATE TABLE IF NOT EXISTS `mytable` (`id` int(10)) ENGINE=InnoDB",
-		"CREATE TABLE IF NOT EXISTS mytable (`id` int(10)) ENGINE=InnoDB",
+		"CREATE TABLE /*generated by server */ mydb.mytable (`id` int(10)) ENGINE=InnoDB",
+		"CREATE TABLE `mydb`.`mytable` (`id` int(10)) ENGINE=InnoDB",
+		"CREATE TABLE IF NOT EXISTS mydb.`mytable` (`id` int(10)) ENGINE=InnoDB",
+		"CREATE TABLE IF NOT EXISTS `mydb`.mytable (`id` int(10)) ENGINE=InnoDB",
 	}
-	table := []byte("mytable")
-	db := []byte("mydb")
+	table := "mytable"
+	db := "mydb"
+	pr := parser.New()
 	for _, s := range cases {
-		m := expCreateTable.FindSubmatch([]byte(s))
-		mLen := len(m)
-		if m == nil || !bytes.Equal(m[mLen-1], table) || (len(m[mLen-2]) > 0 && !bytes.Equal(m[mLen-2], db)) {
-			t.Fatalf("TestCreateTableExp: case %s failed\n", s)
+		stmts, _, err := pr.Parse(s, "", "")
+		if err != nil {
+			t.Fatalf("TestCreateTableExp:case %s failed\n", s)
+		}
+		for _, st := range stmts {
+			nodes := parseStmt(st)
+			if len(nodes) == 0 {
```

Comment:
`if len(nodes) != 1`, then fatal? similar with others.

Reply:
one stmt may contains multi  tables, eg: rename a to b,c to d; then nodes will contain two nodes  a and c.

Reply:
yep, I think in these cases, we must _assert_ the number of the nodes is what we want.

Reply:
what we need to do is to range nodes and then clear old table cache, rather than care about  the number if nodes. if len(nodes)<0 range it, if len(nodes)==0  do nothing.

Reply:
In test cases, we must ensure `parseStmt` works correctly.

Reply:
yep,  some test  had been writed to test `parseStmt`

Reply:
> some test had been writed to test `parseStmt`

so, test the length of return `nodes` too?